File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    String alphabet = "AEIOU";
    int N = alphabet.length();
    String target;
    int cnt = 0;
    boolean found = false;
    
    public void dfs(String w) {
        if (found) return;
        if (w.equals(target)) {
            found = true;
            return;
        }
        if (w.length() == N) return;
        
        for (int i = 0; i < N; i++) {
            cnt += 1;
            dfs(w+alphabet.charAt(i));
            
            if (found) return;
        }
    }
    
    
    public int solution(String word) {
        target = word;
        dfs("");
        return cnt;
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
   
def solution(target):
    alphabet = "AEIOU"
    cnt = 0
    
    def dfs(word) :
        nonlocal cnt
        
        if word == target :
            return True
        
        if len(word) == 5 :
            return False
        
        for a in alphabet :
            cnt += 1
            if dfs(word + a) :
                return True

        return False
            
    dfs("")
    
    return cnt
    
    
    
    
    
